lib/matplotlib/backend_bases.py

```diff
@@ -2952,6 +2952,63 @@ def _zoom_pan_handler(self, event):
             elif event.name == "button_release_event":
                 self.release_zoom(event)
 
+    def _start_event_axes_pan(self, event, method="zoom"):
+        # call "ax.start_pan(..)" on all relevant axes of an event
+
+        def _ax_filter(ax):
+            return (ax.in_axes(event) and
+                    ax.get_navigate() and
+                    getattr(ax, f"can_{method}")()
+                    )
+
+        def _capture_events(ax):
+            f = ax.get_forward_navigation_events()
+            if f == "auto":  # (capture = patch visibility)
+                f = not ax.patch.get_visible()
+            return not f
+
+        # get all relevant axes for the event
+        axes = list(filter(_ax_filter, self.canvas.figure.get_axes()))
+
+        if len(axes) == 0:
+            return
+
+        if self._nav_stack() is None:
+            self.push_current()   # Set the home button to this view.
+
+        # group axes by zorder (reverse to trigger later axes first)
+        grps = dict()
```

Comment:
use a `defaultdict`?

Reply:
The very first implementation of this PR used a `defaultdict` but this was discouraged by @anntzer here: https://github.com/matplotlib/matplotlib/pull/22347#discussion_r795046200

> (one problem with defaultdict is that any __getitem__ could be inserting a new key into the dict, and it's hard to follow which ones do so; using setdefault makes this explicit.)

so I switched to using `dict.setdefault` to address the concerns. 
I'd rather not switch back again except if there's an actual reason why this would be beneficial.

Reply:
Fair enough.  I'm not sure I'm fully convinced, but definitely not worth changing back.